### ISSUE TITLE
Pylab contourf plot using Mollweide projection create artefacts

### ISSUE THREAD

**Issue by ghost** [matplotlib/matplotlib#1910]
matplotlib 1.1.1rc, python 2.7.3, Ubuntu 12.04 LTS

pylab.contourf create artefacts when used in a Mollweide projection.

Here is an example code:

```
import numpy as np
import pylab as plt

ra = np.linspace(-np.pi, np.pi, 40)
dec= np.linspace(-np.pi/2, np.pi/2, 20)
X,Y = np.meshgrid(ra,dec)
Z = np.sin(X) * np.cos(X) * np.sin(Y) * np.cos(Y)

plt.figure()
ax = plt.subplot(111, projection = 'mollweide')
ax.contourf(X,Y,Z,100)
ax.contour(X,Y,Z,10,colors='k')

plt.show()
```

Some of the points seem to be wrongly drawn (See Figure below).
![mollweide](https://f.cloud.github.com/assets/4168086/384660/20cbddfa-a660-11e2-8daf-16b3e31cddc5.png)
The same code used without projection outputs a perfectly normal map: 
![normal](https://f.cloud.github.com/assets/4168086/384661/2bd0dc0a-a660-11e2-8155-01bfd5bbb82d.png)

You may have noticed that the contour lines work in both images. The artefacts seem to be always at the same place.

[Originally posted on Stackoverflow: http://stackoverflow.com/questions/16018243/pylab-contour-plot-using-mollweide-projection-create-artefacts ]

**Comment by dstansby** [comment]
Looks a bit better now! This is on 2.0.0b4+2949.gf12abca

![figure_1](https://cloud.githubusercontent.com/assets/6197628/20644240/073f07c2-b426-11e6-9cce-1429307bb945.png)

**Comment by tacaswell** [comment]
Fixed as far back as 1.5.0!

Might be interesting to track down when this was fixed (git bisect is fun!), but closing.

**Comment by tacaswell** [comment]
![so](https://cloud.githubusercontent.com/assets/199813/20650915/ea7201c6-b4a7-11e6-8de7-895e0c742a9f.png)

```python
import numpy as np
import matplotlib.pyplot as plt
import matplotlib

plt.ion()

ra = np.linspace(-np.pi, np.pi, 40)
dec= np.linspace(-np.pi/2, np.pi/2, 20)
X,Y = np.meshgrid(ra,dec)
Z = np.sin(X) * np.cos(X) * np.sin(Y) * np.cos(Y)

plt.figure()
ax = plt.subplot(111, projection = 'mollweide')
ax.contourf(X,Y,Z,100)
ax.contour(X,Y,Z,10,colors='k')
ax.set_title(matplotlib.__version__)
plt.show()
```
